src/uwtools/tests/test_schemas.py

```diff
@@ -774,6 +779,37 @@ def test_schema_sfc_climo_gen_run_dir(sfc_climo_gen_prop):
     assert "88 is not of type 'string'" in errors(88)
 
 
+# shave
+
+
+def test_schema_shave():
+    config = {
+        "config": {
+            "input_grid_file": "/path/to/input_grid_file",
+            "nx": 88,
+            "ny": 88,
+            "nh4": 1,
+        },
+        "execution": {"executable": "shave"},
+        "run_dir": "/tmp",
+    }
+    errors = schema_validator("shave", "properties", "shave")
+    # Basic correctness:
+    assert not errors(config)
+    # Some top-level keys are required:
+    for key in ("execution", "run_dir", "config"):
+        assert f"'{key}' is a required property" in errors(with_del(config, key))
+    # Additional top-level keys are not allowed:
+    assert "Additional properties are not allowed" in errors({**config, "foo": "bar"})
+
```

Comment:
You could add a test here to confirm that the values under `config` need to be of a specific type as well.

Similar to what exists for `esg-grid`.

i.e. 
```
assert "not of type 'number'" in errors({"config": {"nx": "/some/str"}})
```